libs/blocks/recommended-articles/recommended-articles.js

```diff
@@ -0,0 +1,105 @@
+import { createTag, getMetadata, getConfig, decoratePlaceholders } from '../../utils/utils.js';
+import fetchTaxonomy from '../../scripts/taxonomy.js';
+
+async function getArticleDetails(article) {
+  const path = new URL(article.href).pathname;
+  const resp = await fetch(path);
+  if (!resp || !resp.ok) {
+    // eslint-disable-next-line no-console
+    console.log(`Could not retrieve metadata for ${path}`);
+    return null;
+  }
+  const html = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+
+  let title = getMetadata('og:title', doc).trim();
+  const trimEndings = ['|Adobe', '| Adobe', '| Adobe Blog', '|Adobe Blog'];
+  trimEndings.forEach((ending) => {
+    if (title.endsWith(ending)) title = title.substr(0, title.length - ending.length);
+  });
+
+  return {
+    title,
+    path,
+    category: getMetadata('article:tag', doc),
+    description: getMetadata('description', doc),
+    imageEl: doc.querySelector('picture'),
+    date: getMetadata('publication-date', doc),
+  };
+}
+
+function getDecoratedCards(articles, taxonomy) {
+  const container = createTag('div', { class: 'article-cards' });
+  articles.forEach((article) => {
+    if (!article) return;
+
+    const {
+      category, imageEl, path, title, description, date,
+    } = article;
+
+    const wrapper = createTag('a', { class: 'article-card', href: path });
+    const cardImage = createTag('div', { class: 'article-card-image' }, imageEl);
+    const cardBody = createTag('div', { class: 'article-card-body' });
+
+    const categoryTaxonomy = taxonomy.get(category || 'News');
+    const categoryLink = createTag('a', { href: categoryTaxonomy.link }, categoryTaxonomy.name);
+    const categoryEl = createTag('p', { class: 'article-card-category' }, categoryLink);
+
+    const titleEl = createTag('h3', null, title);
+
+    const descriptionEl = createTag('p', { class: 'article-card-description' }, description);
+
+    const dateEl = createTag('p', { class: 'article-card-date' }, date);
```

Comment:
No new lines between variables
```suggestion
    const {
      category, imageEl, path, title, description, date,
    } = article;
    const wrapper = createTag('a', { class: 'article-card', href: path });
    const cardImage = createTag('div', { class: 'article-card-image' }, imageEl);
    const cardBody = createTag('div', { class: 'article-card-body' });
    const categoryTaxonomy = taxonomy.get(category || 'News');
    const categoryLink = createTag('a', { href: categoryTaxonomy.link }, categoryTaxonomy.name);
    const categoryEl = createTag('p', { class: 'article-card-category' }, categoryLink);
    const titleEl = createTag('h3', null, title);
    const descriptionEl = createTag('p', { class: 'article-card-description' }, description);
    const dateEl = createTag('p', { class: 'article-card-date' }, date);
```